backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/S3Service.java

```diff
@@ -33,46 +46,159 @@
   private static final Logger LOGGER = LogManager.getLogger(S3Service.class);
   private AmazonS3 s3Client;
   private String bucketName;
+  private final Regions awsRegion;
 
-  public S3Service(String cloudBucketName) {
+  public S3Service(String cloudBucketName) throws ModelDBException {
     App app = App.getInstance();
     String cloudAccessKey = app.getCloudAccessKey();
     String cloudSecretKey = app.getCloudSecretKey();
     String minioEndpoint = app.getMinioEndpoint();
-    final Regions awsRegion = Regions.fromName(app.getAwsRegion());
+    awsRegion = Regions.fromName(app.getAwsRegion());
+    this.bucketName = cloudBucketName;
+
     if (cloudAccessKey != null && cloudSecretKey != null) {
       if (minioEndpoint == null) {
-        BasicAWSCredentials awsCreds = new BasicAWSCredentials(cloudAccessKey, cloudSecretKey);
-        this.s3Client =
-            AmazonS3ClientBuilder.standard()
-                .withRegion(awsRegion)
-                .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
-                .build();
+        LOGGER.debug("config based credentials based s3 client");
+        initializeS3ClientWithAccessKey(cloudAccessKey, cloudSecretKey, awsRegion);
       } else {
-
-        AWSCredentials awsCreds = new BasicAWSCredentials(cloudAccessKey, cloudSecretKey);
-        ClientConfiguration clientConfiguration = new ClientConfiguration();
-        clientConfiguration.setSignerOverride("AWSS3V4SignerType");
-
-        this.s3Client =
-            AmazonS3ClientBuilder.standard()
-                .withEndpointConfiguration(
-                    new AwsClientBuilder.EndpointConfiguration(minioEndpoint, app.getAwsRegion()))
-                .withPathStyleAccessEnabled(true)
-                .withClientConfiguration(clientConfiguration)
-                .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
-                .build();
+        LOGGER.debug("minio client");
+        initializeMinioClient(cloudAccessKey, cloudSecretKey, awsRegion, minioEndpoint);
       }
+    } else if (ModelDBUtils.isEnvSet(ModelDBConstants.AWS_ROLE_ARN)
+        && ModelDBUtils.isEnvSet(ModelDBConstants.AWS_WEB_IDENTITY_TOKEN_FILE)) {
```

Comment:
is this two env variable always there before starting MDB? if yes then which file it is? If you will add a sample file in the MDB, then it would be easy to understand that files.

Reply:
no, they will  not always there